lib/Kitten/Monad.hs

```diff
@@ -81,6 +82,8 @@ instance (Monad m) => Monad (KT m) where
     case mx of
       Left reports' -> return $ Left reports'
       Right (x, reports') -> unKT (f x) context reports'
+
+instance Monad m => MonadFail (KT m) where
```

Comment:
We might actually just be able to remove this now. I can do that refactoring if it’s nontrivial, but nothing ought to be depending on it.

I’ve been toying with new ideas in the `ktn` repo though, so I don’t really want to do much maintenance here at the moment.